htdocs/fourn/class/fournisseur.commande.class.php

```diff
@@ -3368,7 +3375,8 @@ public function fetch($rowid)
 		$sql .= ' cd.remise, cd.remise_percent, cd.subprice,';
 		$sql .= ' cd.info_bits, cd.total_ht, cd.total_tva, cd.total_ttc,';
 		$sql .= ' cd.total_localtax1, cd.total_localtax2,';
-		$sql .= ' p.ref as product_ref, p.label as product_label, p.description as product_desc,';
+		$sql .= ' cd.label,';
```

Comment:
This column "label" on line level has been deprecated a long time ago.
The line of products contains a description or a product, and label of product come from the table product.

Reply:
Yes but it's not a good way : if we change the label of the product, it must doesn't change on the supplier order (same thing on the supplier bill)
on the supplier proposal we have the label on the line level and it's a good thing

Reply:
When the reference (id) is the same, changing the label must change the label everywhere. This is the goal of couple ref / label. If ref (primary key) point to a record, changing property for example to fix a typo error, will be visible on all document that point to it. It is the reference that define that it is same or another product, not the label.

Reply:
NO !!!!!! 
the wording of a generated and transmitted document should never change, even if there is a spelling error.
If you order something from a supplier, and you change the wording of your order there could be a huge dispute.

Non, le libellé d'un document généré et transmis ne dois jamais changer, meme si il y a une faute d'orthographe.
Si tu commandes quelque chose à un fournisseur, et que tu change le libellé de ta commande il peu y avoir un énorme litige.

Reply:
Allowing this sort of thing is a monstrous functional error

Reply:
This is true for a description or a ref, less for a short label. By definition a short label is just a clear text of a reference (it is even a different value for each language).
And if you duplicate data for a short label, why not doing it also for the reference (most important than the short label, the width, heigth, size of product). We can duplicate any fields of product into the line of objects but if there is one field that is less important than reference, size, width, color, description, it is surely the short label of a product reference since the reference rules most than label (this default label is even not used in most cases in multi-language context).

Reply:
the subject for me is to have the same operation whether for the parts on the customer side as for the parts on the supplier side
OR the proposals (customers and suppliers) work as well but not the orders and invoices: for the customer part we keep the wording on the line but not for the supplier part

-- 

le sujet pour moi c'est d'avoir le même fonctionnement que ce soit pour les pièces coté clients que pour les pièces coté fournisseurs
OR les propositions (clientes et fournisseurs) fonctionnent ainsi mais pas les commandes et les factures : pour la partie client on conserve le libellé sur la ligne mais pas pour la partie fournisseur